src/module/helpers-treasure.js

```diff
@@ -118,24 +118,26 @@ export async function rollTreasure(table, options = {}) {
       }
     });
   }
-
-  const html = await renderTemplate(
-    `${OSE.systemPath()}/templates/chat/roll-treasure.html`,
-    templateData
-  );
-
-  const chatData = {
-    content: html,
-    // sound: "systems/ose/assets/coins.mp3"
-  };
-
-  const rollMode = game.settings.get("core", "rollMode");
-  if (["gmroll", "blindroll"].includes(rollMode))
-    chatData.whisper = ChatMessage.getWhisperRecipients("GM");
-  if (rollMode === "selfroll") chatData.whisper = [game.user._id];
-  if (rollMode === "blindroll") chatData.blind = true;
-
-  ChatMessage.create(chatData);
+  
+  setTimeout(async () => {
```

Comment:
Would you mind giving this a try before the `renderTemplate` line? This would replace the `setTimeout`.
```js
await new Promise(resolve => requestAnimationFrame(resolve));
```
The 0ms timeout is probably serving to delay execution of the render and chat card code until the rest of the event loop empties out. Waiting for an animation frame should do similar, but without all the mysticism -- instead, we're just straight up asking the browser to let us know when the UI is ready to receive an update.

Reply:
Ok, it's pushed up.  Thanks again.